beacon-chain/db/block.go

```diff
@@ -155,12 +155,12 @@ func (db *BeaconDB) BlockBySlot(slot uint64) (*pb.BeaconBlock, error) {
 
 		blockRoot := mainChain.Get(slotEnc)
 		if blockRoot == nil {
-			return nil
+			return fmt.Errorf("block not found for slot %d", slot)
```

Comment:
Should this be an error? I think returning nil is acceptable if some object is not present in a datastore

Reply:
If we want to return an error, then we should return a package level constant error not found

```
const ErrNotFound = errors.New("not found")

...

if blockRoot == nil {
    return ErrNotFound
}
```

This way you can do special handling like
```
err := db.BlockBySlot(slot)
if err == db.ErrNotFound {
  // handle not found err
  return
}
if err != nil {
  // handle some other error
  log.Error(err)
}
...
```